armi/reactor/converters/uniformMesh.py

```diff
@@ -186,33 +186,45 @@ def makeAssemWithUniformMesh(sourceAssem, newMesh):
             # type to use. This uses the following rules:
-            #     1. Select the first block that has either FUEL or CONTROL flags
-            #     2. Fail if multiple blocks meet this criteria if they have different XS types
-            #     3. Default to the first block in the list if no blocks meet FUEL or CONTROL flags criteria.
+            #     1. Determine the total height corresponding to each XS type that
+            #     appears for blocks with FUEL or CONTROL flags in this domain.
+            #     2. Determine the single XS type that represents the largest fraction
+            #     of the total height of FUEL or CONTROL cross sections.
+            #     3. Use the first block of the majority XS type as the representative block.
+            typeHeight = collections.defaultdict(float)
             blocks = [b for b, _h in overlappingBlockInfo]
-            sourceBlock = None
-            xsType = None
-            for b in blocks:
+            for b, h in overlappingBlockInfo:
                 if b.hasFlags([Flags.FUEL, Flags.CONTROL]):
-                    if sourceBlock is None:
-                        sourceBlock = b
-                        xsType = b.p.xsType
-                    else:
-                        # If there is a duplicate source block candidate that has a different
-                        # cross section type then this is an error because the code cannot
-                        # decide which one is correct.
-                        if b.p.xsType != xsType:
-                            msg = (
-                                f"{sourceBlock} and {b} in {newAssem} have conflicting XS types and are "
-                                f"candidates for the source block. To fix this, either set their XS types "
-                                f"to be the same or remove these flags {[Flags.FUEL, Flags.CONTROL]} "
-                                f"from one of the blocks."
-                            )
-                            runLog.error(msg)
-                            raise ValueError(msg)
-
-            # If no blocks meet the criteria above just select the first block
-            # as the source block and use its cross section type.
+                    typeHeight[b.p.xsType] += h
+
+            sourceBlock = None
+            if len(typeHeight) > 1:
+                # xsType is the one with the majority of overlap
+                xsType = next(
+                    k for k, v in typeHeight.items() if v == max(typeHeight.values())
+                )
+                for b in blocks:
+                    if b.hasFlags([Flags.FUEL, Flags.CONTROL]):
+                        if b.p.xsType == xsType:
+                            sourceBlock = b
+                            break
+
+                if sourceBlock:
+                    totalHeight = sum(typeHeight.values())
+                    runLog.extra(
+                        "Multiple XS types exist between {} and {}".format(
```

Comment:
Just as a note, it might be good to start using f-strings throughout the code base because it can be easier to read and know which parameters are going where. Example: `print(f"This value is: {someVar}")`

No change needed, just a comment.

Reply:
I changed it when I was addressing the other comments